### PR TITLE
feat: crypto.com provider

### PR DESCRIPTION
**Pull request by alexanderbez**
## Changelog

* Implement crypto.com provider
* Import and use `WebsocketController` which is used by the `CryptoProvider`

Note, this was lifted from Umee's version of the price-feeder @ v3.x. When we have sufficient capacity, we may want to explore the generalized improvements both the Kujira and Umee versions have made, such as generalizing the provider implementation.

### CODE REVIEW

File: oracle/price-feeder/oracle/provider/crypto.go [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+	}
+)
+
+func NewCryptoProvider(
+	ctx context.Context,
+	logger zerolog.Logger,
+	endpoint config.ProviderEndpoint,
+	pairs ...types.CurrencyPair,
+) (*CryptoProvider, error) {
+	if endpoint.Name != config.ProviderCrypto {
```
Comment (philipsu522) [inline oracle/price-feeder/oracle/provider/crypto.go:111]: shoudl we error here? what is the reasoning behind setting it to crypto.com if it was misconfigured?

**Review by yzang2019 — COMMENTED** [review]
Can you modify config.example.toml to add crypto as well?

**Review by philipsu522 — APPROVED** [review]

File: oracle/price-feeder/oracle/provider/crypto.go (hunk 1 above)
Comment (alexanderbez) [reply oracle/price-feeder/oracle/provider/crypto.go:111]: So this is how all the other providers are implemented, along with the crypto.com one ported from umee. Although, I do agree it should error instead.

If we do that, we should do it for all providers and create an issue for it :)

**Comment by alexanderbez** [comment]
Updated @yzang2019 :)

File: oracle/price-feeder/config.example.toml [hunk 2]
```diff
@@ -35,11 +35,12 @@ quote = "USD"
 
 [[currency_pairs]]
 base = "SOL"
-chain_denom = "usei"
+chain_denom = "usol"
```
Comment (udpatil) [inline oracle/price-feeder/config.example.toml]: This is usei intentionally to have some moving price feed available on Atlantic 2 for usei

**Review by udpatil — CHANGES_REQUESTED** [review]

File: oracle/price-feeder/config.example.toml (hunk 2 above)
Comment (udpatil) [reply oracle/price-feeder/config.example.toml]: sorry @alexanderbez forgot to update this comment, can we make this change before merging?

**Review by udpatil — APPROVED** [review]

**Pull request merged**